### DISCUSSION TITLE
Question: Unpublish old version automatically?

### DISCUSSION THREAD

**Discussion by pzentner** [microsoft/AL-Go#1557] [Ideas]
When running a deployment into a production environment, the new apps are published and installed and the old versions of the apps are still published, but uninstalled.

Question: Is it possible to automatically unpublish the old versions of those apps when deploying the new versions?

**Comment by buzzwick** [comment]
Would love to see this as a feature or at least an option.

**Comment by gntpet** [comment]
Have the same issue. It's not only for PROD environment, but for QA as well (using S2S token, thus Publish-PerTenantExtensionApps is used)

Installed extensions windows looks crazy :) 
![image](https://github.com/microsoft/AL-Go/assets/16342128/befb53b4-58ec-45d5-a015-3e537f8f62ee)

**Comment by freddydk** [comment]
Currently, there is no functionality in the API to unpublish an app that we can use.
There is an idea on msideas and if anybody has the time, it is possible to create this functionality themselves.

See more here: https://github.com/microsoft/navcontainerhelper/issues/2808

If you are an app-developer, it is probably a fairly small exercise to contribute with this.
